automation/Extension/webext-instrumentation/src/lib/js-instruments.ts

```diff
@@ -546,20 +555,23 @@ export function jsInstruments(event_id, sendMessagesToLogger) {
   ) {
     // Store original descriptor in closure
     const propDesc = Object.getPropertyDescriptor(object, propertyName);
-    if (!propDesc) {
-      console.error(
-        "Property descriptor not found for",
-        objectName,
-        propertyName,
-        object,
-      );
-      return;
-    }
+
+    // Property descriptor for undefined properties
+    let undefinedPropValue;
+    const undefinedPropDesc = {
+      get: () => {
+        return undefinedPropValue;
+      },
+      set: value => {
+        undefinedPropValue = value;
+      },
+      enumerable: false,
+    };
 
     // Instrument data or accessor property descriptors
-    const originalGetter = propDesc.get;
-    const originalSetter = propDesc.set;
-    let originalValue = propDesc.value;
+    const originalGetter = propDesc ? propDesc.get : undefinedPropDesc.get;
```

Comment:
Sure. We can do that in basically two ways as i see it: 1. Require that non-existing properties be instrumented using `nonExistingPropertiesToInstrument` rather than `propertiesToInstrument`. 2. Add a flag `instrumentNonExistingProperties`.

I think 1 is clearer, since it will still fail if non-existing properties are supplied in propertiesToInstrument.